src/containers/MintGenerative/StepExtraSettings/StepExtraSettings.tsx

```diff
@@ -0,0 +1,364 @@
+  // the explore options
+  const [preExploreOptions, setPreExploreOptions] = useState<string>("infinite")
+  const [postExploreOptions, setPostExploreOptions] =
+    useState<string>("infinite")
+
+  // REFERENCES
+  const iframeRef = useRef<ArtworkIframeRef>(null)
+
+  const inputBytes =
+    useMemo<string | null>(() => {
+      if (!data) return null
+      return serializeParams(data, params)
+    }, [data, params]) || state.previewInputBytes
+
+  // DERIVED FROM STATE
+  // the url to display in the iframe
+  const iframeUrl = useMemo<string>(() => {
+    return ipfsUrlWithHashAndParams(state.cidUrlParams!, hash, inputBytes)
+  }, [hash, inputBytes])
+
+  // FUNCTIONS
+  // some setters to update the settings easily
+  const updateExplorationSetting = (
+    target: VariantSettingsTabKey,
+    setting: "enabled" | "hashConstraints" | "paramsConstraints",
+    value: any
+  ) => {
+    setSettings((currentSettings) => ({
+      ...currentSettings,
+      exploration: {
+        ...currentSettings.exploration,
+        [target]: {
+          ...currentSettings.exploration![target],
+          [setting]: value,
+        },
+      },
+    }))
+  }
+
+  // add the current hash to the list of hashes of a mint category
+  const addCurrentVariant = (target: VariantSettingsTabKey) => {
+    const currentHashConstraints =
+      settings.exploration?.[target]?.hashConstraints || []
+    if (!currentHashConstraints.includes(hash)) {
+      currentHashConstraints.push(hash)
+      updateExplorationSetting(
+        target,
+        "hashConstraints",
+        currentHashConstraints
+      )
+      if (usesParams && inputBytes) {
+        const currentParamConstraints =
+          settings.exploration?.[target]?.paramsConstraints || []
+        currentParamConstraints.push(inputBytes)
+        updateExplorationSetting(
+          target,
+          "paramsConstraints",
+          currentParamConstraints
+        )
+      }
+    }
+  }
+
+  // process the user input and turn inputs into a proper settings
+  const onSubmit = (evt: any) => {
+    evt.preventDefault()
+
+    const cloned = cloneDeep(settings)
+    // if explore options are set to infinite, we force hash list to be null
+    if (
+      preExploreOptions === "infinite" &&
+      cloned.exploration?.preMint?.hashConstraints
+    ) {
+      cloned.exploration.preMint.hashConstraints = null
+    }
+    if (
+      preExploreOptions === "infinite" &&
+      cloned.exploration?.preMint?.paramsConstraints
+    ) {
+      cloned.exploration.preMint.paramsConstraints = null
+    }
+    if (
+      postExploreOptions === "infinite" &&
+      cloned.exploration?.postMint?.hashConstraints
+    ) {
+      cloned.exploration.postMint.hashConstraints = null
+    }
+    if (
+      postExploreOptions === "infinite" &&
+      cloned.exploration?.postMint?.paramsConstraints
+    ) {
+      cloned.exploration.postMint.paramsConstraints = null
+    }
```

Comment:
we could have something like

```
if (preExploreOptions === "infinite") {
  const preMint = cloned.exploration?.preMint
  const postMint = clone.exploration?.postMint
  if (preMint?.hashConstraints) cloned.exploration.preMint.hashConstraints = null
  ...
}
```

again, opinionated one, but always easier to understand/debug if there's less branching for me!

Reply:
jeah good point! thanks man!